rpm/centos-10/Dockerfile

```diff
@@ -0,0 +1,35 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_IMAGE=golang:latest
+ARG DISTRO=centos
+ARG SUITE=10
+ARG BUILD_IMAGE=quay.io/centos/${DISTRO}:stream${SUITE}-development
+
+FROM ${GO_IMAGE} AS golang
+
+FROM ${BUILD_IMAGE}
+ENV GOPROXY=https://proxy.golang.org|direct
+ENV GO111MODULE=off
+ENV GOPATH=/go
+ENV GOTOOLCHAIN=local
+ENV PATH=$PATH:/usr/local/go/bin:$GOPATH/bin
+ENV AUTO_GOPATH=1
+ENV DOCKER_BUILDTAGS=exclude_graphdriver_btrfs
+ARG DISTRO
+ARG SUITE
+ENV DISTRO=${DISTRO}
+ENV SUITE=${SUITE}
+
+# RHEL8 / CentOS 8 changed behavior and no longer "rpm --import" or
+# "rpmkeys --import"as part of rpm package's %post scriplet. See
+# https://forums.centos.org/viewtopic.php?f=54&t=72574, and
+# https://access.redhat.com/solutions/3720351
+RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
+RUN dnf install -y rpm-build rpmlint dnf-plugins-core
+RUN dnf config-manager --set-enabled crb
```

Comment:
Looks like some changes are needed;

```

 > [stage-1 2/8] RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial:
0.180 error: /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial: import read failed(2).
------
Dockerfile:27
--------------------
  25 |     # https://forums.centos.org/viewtopic.php?f=54&t=72574, and
  26 |     # https://access.redhat.com/solutions/3720351
  27 | >>> RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial
  28 |     RUN dnf install -y rpm-build rpmlint dnf-plugins-core
  29 |     RUN dnf config-manager --set-enabled crb
--------------------
ERROR: failed to solve: process "/bin/sh -c rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centosofficial" did not complete successfully: exit code: 1
make[1]: *** [Makefile:97: centos-10] Error 1
make[1]: Leaving directory '/home/ubuntu/workspace/docker-ce-packaging_PR-1104/rpm'
make: *** [Makefile:79: centos-10] Error 2
script returned exit code 2
```

Reply:
```bash
docker run --rm quay.io/centos/centos:stream10-development ls -l /etc/pki/rpm-gpg/
total 8
-rw-r--r-- 1 root root 2182 Sep 30 20:00 RPM-GPG-KEY-CentOS-SIG-Extras-SHA512
-rw-r--r-- 1 root root 1683 Sep 30 20:00 RPM-GPG-KEY-centosofficial-SHA256
```

Reply:
One more failure; looks like `rpmlint` is no more for CentOS Stream 10 ("CS10");

- https://issues.redhat.com/browse/CS-2451
- https://issues.redhat.com/browse/CS-2453